packages/chain-adapters/src/utxo/utxoSelect/index.ts

```diff
@@ -64,17 +64,17 @@ export const utxoSelect = (input: UTXOSelectInput) => {
     return coinSelect<unchained.bitcoin.Utxo>(utxos, outputs, Number(input.satoshiPerByte))
   })()
 
-  /**
-   * note - the 0-indexed output is the `to` address, and if `input.from` is included as an argument, the
-   * 1-indexed output will be the change address, which we overwrite below
-   */
-  if (input.from && result?.outputs?.[1]?.value) {
+  // Finds the change output index (if present), so we can replace it with the from address in case from is provided
+  const changeOutputIndex = (result.outputs ?? []).findIndex(
+    o => o.value && !o.address && !o.script,
+  )
+  if (input.from && result.outputs && changeOutputIndex !== -1) {
```

Comment:
can get rid of the extra check if we want
```suggestion
  if (input.from && changeOutputIndex !== -1) {
```

Reply:
As discussed, we need this or TS will be sad :(